File: nim_core_v2/ios/Classes/services/FLTClientAntispamUtil.swift
// Copyright (c) 2022 NetEase, Inc. All rights reserved.
// Use of this source code is governed by a MIT license that can be
// found in the LICENSE file.

import NIMSDK

enum ClientAntispamUtilType: String {
  case checkTextAntispam
}

class FLTClientAntispamUtil: FLTBaseService, FLTService {
  func serviceName() -> String {
    ServiceType.ClientAntispamUtil.rawValue
  }

  func onMethodCalled(_ method: String, _ arguments: [String: Any],
                      _ resultCallback: ResultCallback) {
    switch method {
    case ClientAntispamUtilType.checkTextAntispam.rawValue:
      checkTextAntispam(arguments, resultCallback)
    default:
      resultCallback.notImplemented()
    }
  }

  func checkTextAntispam(_ arguments: [String: Any], _ resultCallback: ResultCallback) {
    guard let text = arguments["text"] as? String else {
      parameterError(resultCallback)
      return
    }
    let replace = arguments["replace"] as? String
    let result = V2NIMClientAntispamUtil.checkTextAntispam(text, replace: replace)
    successCallBack(resultCallback, result.toDic())
  }

  func register(_ nimCore: NimCore) {
    self.nimCore = nimCore
    nimCore.addService(self)
  }
}

File: nim_core_v2/android/src/main/kotlin/com/netease/nimflutter/services/FLTClientAntispamUtil.kt
/*
 * Copyright (c) 2022 NetEase, Inc. All rights reserved.
 * Use of this source code is governed by a MIT license that can be
 * found in the LICENSE file.
 */

package com.netease.nimflutter.services

import android.content.Context
import com.netease.nimflutter.FLTService
import com.netease.nimflutter.NimCore
import com.netease.nimflutter.NimResult
import com.netease.nimflutter.extension.toMap
import com.netease.nimlib.sdk.v2.utils.V2NIMClientAntispamUtil

class FLTClientAntispamUtil(
    applicationContext: Context,
    nimCore: NimCore
) : FLTService(applicationContext, nimCore) {
    override val serviceName = "V2NIMClientAntispamUtil"

    init {
        registerFlutterMethodCalls(
            "checkTextAntispam" to ::checkTextAntispam
        )
    }

    private suspend fun checkTextAntispam(arguments: Map<String, *>): NimResult<Map<String, Any?>?> {
        val text = arguments["text"] as? String ?: return NimResult(
            code = -1,
            errorDetails = "text is null"
        )
        val replace = arguments["replace"] as? String
        val result = V2NIMClientAntispamUtil.checkTextAntispam(text, replace)
        return NimResult(code = 0, data = result?.toMap())
    }
}
